File: amazia-console/app/Sku/Controller/CreateProductSkuController.php
<?php

namespace App\Sku\Controller;

use App\Sku\Service\CreateProductSkuService;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;
use Illuminate\Routing\Controller;

class CreateProductSkuController extends Controller
{
    public function __construct(private CreateProductSkuService $createProductSkuService) {}

    public function __invoke(Request $request, int $id): JsonResponse
    {
        $response = $this->createProductSkuService->create($id, $request->all());
        return response()->json($response->json(), $response->status());
    }
}

File: amazia-core/src/main/java/com/example/sku/controller/CreateProductSkuController.java
package com.example.sku.controller;

import com.example.sku.entity.ProductSku;
import com.example.sku.service.CreateProductSkuService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class CreateProductSkuController {

    private final CreateProductSkuService createProductSkuService;

    public CreateProductSkuController(CreateProductSkuService createProductSkuService) {
        this.createProductSkuService = createProductSkuService;
    }

    @PostMapping("/products/{id}/skus")
    public ResponseEntity<ProductSku> create(@PathVariable Long id, @RequestBody ProductSku request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(createProductSkuService.create(id, request));
    }
}
